muxers/mplex/src/lib.rs

```diff
@@ -85,25 +85,34 @@ where
     type Substream = Substream<C>;
     type Error = io::Error;
 
-    fn poll_inbound(&self, cx: &mut Context<'_>) -> Poll<Result<Self::Substream, Self::Error>> {
+    fn poll_inbound(
+        self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Result<Self::Substream, Self::Error>> {
         self.io
             .lock()
```

Comment:
Couldn't we now get rid of the `Mutex` in `self.io`?

Reply:
Unfortunately no because we need to share the struct between the substreams and the `StreamMuxer` implementation.